File: juneleetcodeProblem/Tree/treeTRaversal/Inorder.py
class Node:
    def __init__(self,data):
        self.data=data
        self.left=None
        self.right=None
        
    def printInorder(self,root):
        '''
        input :1,2,3,4,5,6,7
        o/p: 4,2,5,1,6,3,7
        '''
        if root:
            # print left part first
            
            self.printInorder(self,root.left)
            print(root.data)
            self.printInorder(self,root.right)
     
    def printPreOrder(self,root):
        if root:
            print(root.data)
            self.printPreOrder(self,root.left)
            self.printPreOrder(self,root.right)
    def printPosrorder(self,root):
        if root:
            self.printPosrorder(self,root.left)
            self.printPosrorder(self,root.right)
            print(root.data) 
     
     
     
     
     
     
     
     
     
     
     
     
     
     
     
     
     
     
     
     
root=Node(1)
root.left=Node(2)
root.right=Node(3)
root.left.left=Node(4)
root.left.right=Node(5)
root.right.left=Node(6)
root.right.right=Node(7)
print ("Inorder traversal of binary tree is")
Node.printInorder(Node,root)
print ("preorder traversal of binary tree is")
Node.printPreOrder(Node,root)
print ("Postorder traversal of binary tree is")
Node.printPosrorders(Node,root)

File: juneleetcodeProblem/Tree/treeTRaversal/Inorder.java
package juneleetcodeProblem.Tree.treeTRaversal;

class Node {
    int data;
    Node left;
    Node right;

    public Node(int item) {
        this.data = item;
        this.left = null;
        this.right = null;

    }

}

public class Inorder {
    Node root;

    public void printInorder(Node root) {
        if (root == null)
            return;

        /* first recur on left child */
        printInorder(root.left);

        /* then print the data of node */
        System.out.print(root.data + " ");

        /* now recur on right child */
        printInorder(root.right);
    }

    public void printPreorder(Node root) {
        if (root == null) {
            return;

        }
        System.out.print(root.data + " ");
        printPreorder(root.left);
        printPreorder(root.right);
    }

    public void PrintpostOrder(Node root) {
        if (root == null) {
            return;

        }
        PrintpostOrder(root.left);
        PrintpostOrder(root.right);
        System.out.print(root.data + " ");
    }

    public static void main(String[] args) {
        Inorder tree = new Inorder();
        tree.root = new Node(1);
        tree.root.left = new Node(2);
        tree.root.right = new Node(3);
        tree.root.left.left = new Node(4);
        tree.root.left.right = new Node(5);
        tree.root.right.left = new Node(6);
        tree.root.right.right = new Node(7);
        System.out.println("\nInorder traversal of binary tree is ");
        tree.printInorder(tree.root);

        System.out.println("\nPreorder traversal of binary tree is ");
        tree.printPreorder(tree.root);

        System.out.println("\npostorder traversal of binary tree is ");
        tree.PrintpostOrder(tree.root);

    }

}
